src/mame/efo/cedar_magnet.cpp

```diff
@@ -265,6 +265,440 @@ MASTER BOARD
    |                                                         __________     |
    |                                                        |MC74HC157|     |
    |________________________________________________________________________|
+*/
+
+/*
+Additional hardware notes from ArcadeHacker:
+
+
+Z80 main @ 3.9986 Mhz
+---------------------
+
+Memory map
+‘Bank bits’ are these from IC 48 PIO  (0x20)
+A5 -> 4 74HC10 3NAND IC19  // Main CPU Memory bank switch
+A6 -> 2 74HC10 3NAND IC19  // Main CPU Memory bank switch
+
+Initial state  (bank bits 11?)
+0000-1FFF eprom
+2000-3FFF eprom mirror
+4000-5FFF eprom mirror
+6000-7FFF eprom mirror
+8000-9FFF eprom mirror
+A000-BFFF eprom mirror
+C000-DFFF eprom mirror
+E000-FFFF eprom mirror
+
+Upload to other CPUs state (Bank bits 10 )
+
+This is complex and not 100% understood.
+
+ 0000 - BFFF   - RAM?  (same as in state 01 / 00)
+ C000  FFFF   - a 0x4000 sized window into the RAM for the other CPUs, uploads programs**
+
+**see 0x43 IC 49 PIO
+B2 -> 36 J7
+B3 -> 35 J7
+and MAYBE  B1 -> 3 74HC04 IC21
+2 bank bits gives a full 0x10000 (64kb view) but could be more?
+
+WHICH cpu we get a window into accessible seems to be determined by the busreq bits (main CPU must have bus?)
+IC48 A1 -> 47 J6 - output (DH: busreq plane board a?)
+IC48 B1 -> 48 J6 - output (DH: busreq plane board b?)
+IC48 B5 -> 43 J6 - output (DH: busreq sprite board?)
+IC49 B5 -> 8 J3  - output (DH: sound CPU busreq)
+
+Is it possible to upload to multiple CPUs at once this way??
+
+
+Simple Ram/ROM state (Banks bits 01)
+     0000 - DFFF   - RAM?    (same as in state 10 / 00)
+     E000-FFFF eprom 
+
+ALL RAM (Banks bits 00)
+     0000 - DFFF   - RAM?    (same as in state 10 / 01)
+     E000-FFFF - more ram, only accessible in this state?
+
+
+I/O map
+
+0x18 <-> 0001 1000 ?
+0x19 <-> 0001 1001 ?
+0x1A ->  0001 1010 ?
+0x1B <-  0001 1011 ?
+
+Loop checking bit 2 on port 19
+// 9496 in a,($19)
+// 9498 bit 2,a
+Writes a stream of data from rom to the other ports (garbage??)
+
+IC 48 PIO 
+0x20 0010 0000 IC48 Data port A    // Bits 5&6 manage banked mem
+0x21 0010 0001 IC48 Command port A
+
+A0 -> 49 J6 - input from plane board a (main CPU checks before accessing)
+A1 -> 47 J6 - output (DH: busreq plane board a?)
+A2 -> 45 J6 - output (DH: reset plane board a CPU?)
+A3 <- 4 74HC14P (inverter) IC4 <- EDGE 21 COIN1
+A4 <- 9 74HC74 IC20 <- input from 18 74LS244 IC61   - Interrupt source related? (DH - gets checked in IRQ code)
+A5 -> 4 74HC10 3NAND IC19  // Main CPU Memory bank switch
+A6 -> 2 74HC10 3NAND IC19  // Main CPU Memory bank switch
+A7 -> 12 J4 unpopulated
+
+0x22 0010 0010 IC48 Data port B 
+0x23 0010 0011 IC48 Command port B
```

Comment:
The binary values for addresses add clutter without really providing value, and they aren’t even used consistently.  Just get rid of them altogether.